sdk/resourcemanager/azure-resourcemanager-keyvault/src/main/java/com/azure/resourcemanager/keyvault/implementation/ManagedHsmsImpl.java

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.keyvault.implementation;
+
+import com.azure.core.http.rest.PagedFlux;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.util.CoreUtils;
+import com.azure.resourcemanager.keyvault.KeyVaultManager;
+import com.azure.resourcemanager.keyvault.fluent.ManagedHsmsClient;
+import com.azure.resourcemanager.keyvault.fluent.models.ManagedHsmInner;
+import com.azure.resourcemanager.keyvault.models.ManagedHsm;
+import com.azure.resourcemanager.keyvault.models.ManagedHsmProperties;
+import com.azure.resourcemanager.keyvault.models.ManagedHsms;
+import com.azure.resourcemanager.resources.fluentcore.arm.collection.implementation.GroupableResourcesImpl;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+
+/**
+ * Implementation for ManagedHsms and its parent interfaces.
+ */
+public class ManagedHsmsImpl
+    extends GroupableResourcesImpl<ManagedHsm, ManagedHsmImpl, ManagedHsmInner, ManagedHsmsClient, KeyVaultManager>
+    implements ManagedHsms {
+    private final String tenantId;
+
+    public ManagedHsmsImpl(KeyVaultManager manager, String tenantId) {
+        super(manager.serviceClient().getManagedHsms(), manager);
+        this.tenantId = tenantId;
+    }
+
+    @Override
+    protected ManagedHsmImpl wrapModel(String name) {
+        ManagedHsmInner inner = new ManagedHsmInner().withProperties(new ManagedHsmProperties());
+        inner.properties().withTenantId(UUID.fromString(tenantId));
+        return new ManagedHsmImpl(name, inner, this.manager());
+    }
```

Comment:
If this is not used, prefer to add a comment to it.

Reply:
Sure. There's also another [`createResourceAsync`](https://github.com/Azure/azure-sdk-for-java/pull/34059/commits/78df8c2eaeb10eb68f2337b97b039ec523346ba6#diff-34a77c66f14a3a65169bf14faf1ce0f16bab2cf807e452287887e8c0b755f382R56-R59) from `GroupableResourceImpl`(not exposed through `ManagedHsm` interface)

I'll add comments too.

Reply:
`wrapModel` is used by our internal method `wrapPageAsync`. 

Added comment to `createResourceAsync`.

Reply:
https://github.com/Azure/azure-sdk-for-java/pull/34059/files#r1163639388